aip/general/0162.md

```diff
@@ -27,199 +27,174 @@ to the version of an API as a whole.
+referring to specific revision of a resource, the subcollection name **must**
+named `revisions`. Resource revision has the name with the format
+ `{resource_name}/revisions/{revision_id}`. For example:
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+publishers/123/books/les-miserables/revisions/c7cfa2a8
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+### Creating revisions
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-### Getting a revision
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+APIs **may** use any of these strategies. APIs that want to create a revision on
+user request should handle this with a custom method on the resource:
 
 ```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
+rpc CommitBookRevision(CommitBookRevisionRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*}:commitRevision"
+    body: "*"
+  };
+}
+
+message CommitBookRequest {
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
       type: "library.googleapis.com/Book"
     }];
+  string tag = 2 [(google.api.field_behavior) = OPTIONAL];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a long running operation or resource revision.
+- The request message **must** include the `name` field.
+- The resource ID of the resource revision **should** be a string and contain a
+  short, automatically-generated random string. A good rule of thumb is the last
+  eight characters of a [UUID version 4 (random)](UUID4).
+- `tag` field is optional. If provided, revision created will have the tag set.
```

Comment:
since tag is mentioned here, I think the `tagging revisions` section should probably move up so the reader is familiar with the concept before arriving here.